spidermon/contrib/scrapy/monitors/monitors.py

```diff
@@ -557,26 +557,30 @@ def run(self, result):
     def _get_jobs(self, states, number_of_jobs):
         tags = self._get_tags_to_filter()
 
-        jobs = []
+        total_jobs = []
         start = 0
         client = Client(self.crawler.settings)
+        MAX_API_COUNT = 1000
 
-        _jobs = client.spider.jobs.list(
-            start=start,
-            state=states,
-            count=number_of_jobs,
-            filters=dict(has_tag=tags) if tags else None,
-        )
-        while _jobs:
-            jobs.extend(_jobs)
-            start += 1000
-            _jobs = client.spider.jobs.list(
+        while True:
+            # API has a 1000 results limit
+            count = min(number_of_jobs - len(total_jobs), MAX_API_COUNT)
+            current_jobs = client.spider.jobs.list(
                 start=start,
                 state=states,
-                count=number_of_jobs,
+                count=count,
                 filters=dict(has_tag=tags) if tags else None,
```

Comment:
**issue (non-blocking):** This was already like this, so we might need to update it in another PR, but `filters=dict(has_tag=tags) if tags else None` doesn't do anything, at least in the version of `scrapinghub` I'm using (`2.4.0`). `has_tag=tags` does filter the jobs, though I'm not sure if that's backwards compatible.

Reply:
Is there any other issues about this?

Reply:
Is there any other issues about this?

Reply:
No that I know of, no! And that issue is not related to these changes. We should be good to merge 👍